protocols/servlet/src/main/java/org/jboss/arquillian/protocol/servlet/ServletExtension.java

```diff
@@ -32,6 +34,7 @@
    public void register(ExtensionBuilder builder)
    {
       builder.service(Protocol.class, org.jboss.arquillian.protocol.servlet.v_2_5.ServletProtocol.class)
-             .service(Protocol.class, org.jboss.arquillian.protocol.servlet.v_3.ServletProtocol.class);
+             .service(Protocol.class, org.jboss.arquillian.protocol.servlet.v_3.ServletProtocol.class)
+             .service(ResourceProvider.class, ServletContextResourceProvider.class);
```

Comment:
Think you need to add a Validate.classExists("javax.servlet.ServletContext") around the ServletContextResourceProvider service registration to avoid requiring the servlet-api classes to be available on client classpath.